atom/browser/common_web_contents_delegate.cc

```diff
@@ -274,16 +274,20 @@ void CommonWebContentsDelegate::DevToolsAppendToFile(
                  base::Unretained(this), url));
 }
 
-void CommonWebContentsDelegate::DevToolsAddFileSystem() {
-  file_dialog::Filters filters;
-  base::FilePath default_path;
-  std::vector<base::FilePath> paths;
-  int flag = file_dialog::FILE_DIALOG_OPEN_DIRECTORY;
-  if (!file_dialog::ShowOpenDialog(owner_window(), "", default_path,
-                                   filters, flag, &paths))
-    return;
+void CommonWebContentsDelegate::DevToolsAddFileSystem(
+    const base::FilePath& path, const base::Closure& callback) {
+  if (path.empty()) {
+    file_dialog::Filters filters;
+    base::FilePath default_path;
+    std::vector<base::FilePath> paths;
+    int flag = file_dialog::FILE_DIALOG_OPEN_DIRECTORY;
+    if (!file_dialog::ShowOpenDialog(owner_window(), "", default_path,
+                                    filters, flag, &paths))
+      return;
+
+    const_cast<base::FilePath&>(path) = paths[0];
```

Comment:
It's not a good practice to modify `const` argument. You can create a new local one.